polygeist/tools/cgeist/Lib/CGCall.cc

```diff
@@ -1257,14 +1182,10 @@ MLIRScanner::emitGPUCallExpr(clang::CallExpr *Expr) {
 
         if (isa<LLVM::LLVMPointerType>(Arg.getType())) {
           const clang::FunctionDecl *Callee = EmitCallee(Expr->getCallee());
-          LLVM::LLVMFuncOp StrcmpF =
-              Glob.getOrCreateLLVMFunction(Callee, FuncContext);
-          Builder.create<LLVM::CallOp>(
-              Loc, StrcmpF,
-              ValueRange({Builder.create<LLVM::BitcastOp>(
-                  Loc,
-                  Glob.getTypes().getPointerType(Builder.getIntegerType(8)),
-                  Arg)}));
+          FunctionToEmit FTE(*Callee, FuncContext);
+          auto StrcmpF =
+              static_cast<func::FuncOp>(Glob.getOrCreateMLIRFunction(FTE));
```

Comment:
I see other places we use `cast` instead of `static_cast`. Should we also use `cast` here for consistency?

Reply:
Done!